### PR TITLE
Add account migration pallet

### PR DESCRIPTION
**Pull request by AsceticBear**
Part of the  #80

### CODE REVIEW

File: runtime/crab/src/lib.rs [hunk 1]
```diff
@@ -276,6 +276,7 @@ frame_support::construct_runtime! {
 		Assets: pallet_assets = 34,
 		Balances: pallet_balances = 5,
 		TransactionPayment: pallet_transaction_payment = 6,
+		Migrate: darwinia_migrate = 40,
```
Comment (jiguantong) [inline runtime/crab/src/lib.rs:279]: Index conflict with `darwinia_deposit`, xavier/staking branch

File: runtime/pangolin/src/lib.rs [hunk 2]
```diff
@@ -272,6 +272,7 @@ frame_support::construct_runtime! {
 		Assets: pallet_assets = 34,
 		Balances: pallet_balances = 5,
 		TransactionPayment: pallet_transaction_payment = 6,
+		Migrate: darwinia_migrate = 36,
```
Comment (jiguantong) [inline runtime/pangolin/src/lib.rs:275]: I think we should keep this index consistent with other runtimes, although it does not have bridge-related pallets, what do you think? @AurevoirXavier

File: runtime/crab/src/lib.rs (hunk 1 above)
Comment (AsceticBear) [reply runtime/crab/src/lib.rs:279]: I'm not sure which PR will be merged first. So please pay attention to the other parts. We can fix conflicts later.

File: runtime/pangolin/src/lib.rs (hunk 2 above)
Comment (AurevoirXavier) [reply runtime/pangolin/src/lib.rs:275]: We can adjust this later.

**Review by HackFisher — DISMISSED** [review]
Please resolve the conflicts

**Comment by AsceticBear** [comment]
> Please resolve the conflicts

Fixed. @HackFisher @AurevoirXavier

**Comment by AurevoirXavier** [comment]
Suggest renaming to `account-migration`

**Pull request merged**